### PR TITLE
bpo-35397: Remove deprecation and document urllib.parse.unwrap

### PR DESCRIPTION
**Pull request by remilapeyre**
https://bugs.python.org/issue35397

### CODE REVIEW

File: Doc/library/urllib.parse.rst [hunk 1]
```diff
@@ -331,6 +331,13 @@ or on combining URL components into a URL string.
    .. versionchanged:: 3.2
       Result is a structured object rather than a simple 2-tuple.
 
+.. function:: unwrap(url)
+
+   Extract the url from a wrapped URL (that is, a string formatted as
+   ``<URL:type://host/path>``, ``<type://host/path>``, ``URL:type://host/path``
+   or ``type://host/path``). If *url* is not a wrapped URL, it is returned
```
Comment (merwok) [inline Doc/library/urllib.parse.rst]: Minor: is `type` used anywhere else in this doc page?  If not, I think the canonical term `scheme` would be clearer.

**Review by ghost — COMMENTED** [review]
Ok.Open trigger

File: Doc/library/urllib.parse.rst (hunk 1 above)
Comment (remilapeyre) [reply Doc/library/urllib.parse.rst]: `type://` is used nowhere else and is a remnant from the docstring.

File: Lib/urllib/parse.py [hunk 2]
```diff
@@ -952,17 +952,12 @@ def _to_bytes(url):
...
 
 def unwrap(url):
-    warnings.warn("urllib.parse.unwrap() is deprecated as of 3.8",
-                  DeprecationWarning, stacklevel=2)
-    return _unwrap(url)
-
-
-def _unwrap(url):
-    """unwrap('<URL:type://host/path>') --> 'type://host/path'."""
+    """unwrap('<URL:scheme://host/path>') --> 'scheme://host/path'."""
```
Comment (merwok) [inline Lib/urllib/parse.py]:
```suggestion
    """Transform a string like '<URL:scheme://host/path>' into 'scheme://host/path'.
    
    The string is returned unchanged if it's not a wrapped URL.
    """
```

**Review by merwok — APPROVED** [review]

**Comment by csabella** [comment]
@remilapeyre, please apply the change suggested by @merwok.  Thanks!

**Comment by remilapeyre** [comment]
Thanks @csabella, it's done !

File: Lib/urllib/parse.py [hunk 3]
```diff
@@ -952,17 +952,15 @@ def _to_bytes(url):
...
 def unwrap(url):
-    warnings.warn("urllib.parse.unwrap() is deprecated as of 3.8",
-                  DeprecationWarning, stacklevel=2)
-    return _unwrap(url)
-
-
-def _unwrap(url):
-    """unwrap('<URL:type://host/path>') --> 'type://host/path'."""
+    """Transform a string like '<URL:scheme://host/path>' into 'scheme://host/path'.
+
```
Comment (ZackerySpytz) [inline Lib/urllib/parse.py]: This line has trailing whitespace.

**Comment by csabella** [comment]
@remilapeyre, the builds are failing because of the new documentation you added to the `.rst` file for `unwrap`.  It is seeing it as 'suspicious' markup.  Here is the output of the failure:

```
Warning, treated as error:
[library/urllib.parse:379] ":scheme" found in "<URL:scheme://host/path>"
Makefile:44: recipe for target 'build' failed
make[1]: *** [build] Error 2
make[1]: Leaving directory '/home/travis/build/python/cpython/Doc'
Suspicious check complete; look for any errors in the above output or in build/suspicious/suspicious.csv.  If all issues are false positives, append that file to tools/susp-ignored.csv.
Makefile:105: recipe for target 'suspicious' failed
```

So, based on the suggestion there, you need to edit `susp-ignored.csv` in `https://github.com/python/cpython/tree/master/Doc/tools`.  Here's a [link](https://github.com/python/cpython/blob/master/Doc/tools/extensions/suspicious.py) to the checks, for more info.

**Comment by remilapeyre** [comment]
Thanks @csabella, this is now fixed.

**Comment by csabella** [comment]
@remilapeyre, thank you for the contribution!  @merwok, thank you for the review!  :tada:

**Pull request merged**